api/v1alpha2/istio_merge.go

```diff
@@ -40,13 +51,42 @@ func (i *Istio) MergeInto(op iopv1alpha1.IstioOperator) (iopv1alpha1.IstioOperat
 
 	externalNameAliasAnnotationFixOp := manageExternalNameAlias(i, mergedResourcesOp)
 
+	if i.Spec.CompatibilityMode {
+		compatibleIop := setCompatibilityEnvs(i, externalNameAliasAnnotationFixOp)
+		return compatibleIop, nil
+	}
+
 	return externalNameAliasAnnotationFixOp, nil
 }
 
 type meshConfigBuilder struct {
 	c *meshv1alpha1.MeshConfig
 }
 
+func setCompatibilityEnvs(_ *Istio, op iopv1alpha1.IstioOperator) iopv1alpha1.IstioOperator {
+	if op.Spec == nil {
+		op.Spec = &v1alpha1.IstioOperatorSpec{}
+	}
+	if op.Spec.Components == nil {
+		op.Spec.Components = &v1alpha1.IstioComponentSetSpec{}
+	}
+	if op.Spec.Components.Pilot == nil {
+		op.Spec.Components.Pilot = &v1alpha1.ComponentSpec{}
+	}
+	if op.Spec.Components.Pilot.K8S == nil {
+		op.Spec.Components.Pilot.K8S = &v1alpha1.KubernetesResourcesSpec{}
+	}
+
+	for k, v := range compatibilityEnvs {
```

Comment:
We are not using the compatibilityVersion during install. This means we always need to manage a list of ENV Vars. As far as I understood we wanted to decouple as far as possible from this Istio internals and use the [compatibilityVersion](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/#easing-upgrades-with-compatibility-versions) instead and additionally allow configuring env vars that are not covered by the compatibilityVersion, e.g. `ISTIO_DELTA_XDS` for version 1.22.

Reply:
Right, I'll adjust to rely mainly on the compatibilityVersion